EFCore.CheckConstraints/Internal/ValidationCheckConstraintConvention.cs

```diff
@@ -39,7 +72,7 @@ public class ValidationCheckConstraintConvention : IModelFinalizingConvention
             _databaseProvider = databaseProvider;
             _intTypeMapping = relationalTypeMappingSource.FindMapping(typeof(int));
 
-            _supportsRegex = SupportsRegex;
+            _supportsRegex = options.UseRegex && SupportsRegex;
```

Comment:
Let's rename from `_supportsRegex` (which is no longer about database support) to `_useRegex` (which should still combine `options.UseRegex` and `SupportsRegex` as you've done)

Reply:
done.